demo/executor/src/lib.rs

```diff
@@ -18,44 +18,58 @@
-extern crate substrate_runtime_io as runtime_io;
-extern crate substrate_primitives as primitives;
+#[macro_use]
+extern crate substrate_executor;
 extern crate demo_primitives;
 extern crate ed25519;
+extern crate substrate_codec as codec;
+extern crate substrate_primitives as primitives;
+extern crate substrate_runtime_io as runtime_io;
+extern crate substrate_state_machine as state_machine;
 extern crate triehash;
 
-#[cfg(test)] extern crate substrate_keyring as keyring;
-#[cfg(test)] extern crate substrate_runtime_primitives as runtime_primitives;
-#[cfg(test)] extern crate substrate_runtime_support as runtime_support;
-#[cfg(test)] extern crate substrate_runtime_staking as staking;
-#[cfg(test)] extern crate substrate_runtime_system as system;
-#[cfg(test)] #[macro_use] extern crate hex_literal;
+#[cfg(test)]
+extern crate substrate_keyring as keyring;
+#[cfg(test)]
+extern crate substrate_runtime_primitives as runtime_primitives;
+#[cfg(test)]
+extern crate substrate_runtime_staking as staking;
+#[cfg(test)]
+extern crate substrate_runtime_support as runtime_support;
+#[cfg(test)]
+extern crate substrate_runtime_system as system;
+#[cfg(test)]
+#[macro_use]
+extern crate hex_literal;
 
 native_executor_instance!(pub Executor, demo_runtime::api::dispatch, demo_runtime::VERSION, include_bytes!("../../runtime/wasm/target/wasm32-unknown-unknown/release/demo_runtime.compact.wasm"));
 
 #[cfg(test)]
 mod tests {
-	use runtime_io;
 	use super::Executor;
-	use substrate_executor::WasmExecutor;
-	use codec::{Slicable, Joiner};
+	use codec::{Joiner, Slicable};
+	use demo_primitives::{AccountId, BlockNumber, Hash};
+	use demo_runtime::{
+		BareExtrinsic, Block, BuildStorage, Call, Concrete, Extrinsic, GenesisConfig, Header,
+		SessionConfig, Staking, StakingConfig, UncheckedExtrinsic,
+	};
+	use ed25519::{Pair, Public};
 	use keyring::Keyring;
-	use runtime_support::{Hashable, StorageValue, StorageMap};
-	use state_machine::{CodeExecutor, TestExternalities};
 	use primitives::twox_128;
-	use demo_primitives::{Hash, BlockNumber, AccountId};
+	use runtime_io;
 	use runtime_primitives::traits::Header as HeaderT;
-	use runtime_primitives::{ApplyOutcome, ApplyError, ApplyResult, MaybeUnsigned};
+	use runtime_primitives::{ApplyError, ApplyOutcome, ApplyResult, MaybeUnsigned};
+	use runtime_support::{Hashable, StorageMap, StorageValue};
+	use state_machine::{CodeExecutor, TestExternalities};
+	use substrate_executor::WasmExecutor;
 	use {staking, system};
-	use demo_runtime::{Header, Block, UncheckedExtrinsic, Extrinsic, Call, Concrete, Staking,
-		BuildStorage, GenesisConfig, SessionConfig, StakingConfig, BareExtrinsic};
-	use ed25519::{Public, Pair};
 
-	const BLOATY_CODE: &[u8] = include_bytes!("../../runtime/wasm/target/wasm32-unknown-unknown/release/demo_runtime.wasm");
-	const COMPACT_CODE: &[u8] = include_bytes!("../../runtime/wasm/target/wasm32-unknown-unknown/release/demo_runtime.compact.wasm");
+	const BLOATY_CODE: &[u8] = include_bytes!(
```

Comment:
In practice I'd prefix these lines with `#[cfg_attr(rustfmt, rustfmt_skip)]` as I consider this a valid case of going over the char-limit without problems.

Reply:
I agree that this is a valid case for long lines. I also think having to resort to `#[cfg…` is adding noise that we do not want. Close to being a show-stopper to me. :/

Reply:
well, these lines violate the rules even outside of rustfmt: it does violate the 100char rule, including the tab the second even violates the 120chars rule (134chars), which is formulated quite strictly... so a line like this, therefore, should also be flagged in a review with our current style guide - this isn't even rustfmt specific...

either we don't consider the rules as strict or we break the lines, I don't see any other options. And we don't want the rule to be strict, explicitly stating that we are okay with breaking the rule here, sounds like the proper way to do it. What is your proposal, @dvdplm ?